### PR TITLE
Chore: Attach GitHub Action for CI

### PR DESCRIPTION
**Pull request by lukeed**
Imports the `ci.yml` file I use for any of my modules that deal with the filesystem.

Runs Node 10.x, 12.x, 14.x versions on the three OS options: ubuntu, windows, macOS

The verbose `on:` config is to skip CI when a commit and/or PR only changes the `readme.md`, for example.
AFAIK, there's a shortcut like `on: [push, pull_request]` but _not_ a shortcut for having both events share the same config.

### CODE REVIEW

**Review by benmccann — APPROVED** [review]
i'm not much of a Github Actions expert, but this lgtm

**Comment by benmccann** [comment]
Oops. Looks like it failed:

> Error: Unable to resolve action `actions/setup-node@v2`, unable to find version `v2`

**Comment by lukeed** [comment]
Yeah I'm gunna be playing with it. Wasn't a direct 1:1 copy of what I typically use. Sorry for noise :D

File: .eslintrc [hunk 1]
```diff
@@ -18,7 +18,7 @@
     },
     "extends": "eslint:recommended",
     "parserOptions": {
-        "ecmaVersion": 6,
+        "ecmaVersion": 10,
```
Comment (Conduitry) [inline .eslintrc:21]: Why this change, and why this version in particular? What's the minimum Node version do we want this library to require, and - since there's no build/transpile step - what ES version makes sense to lint as?

Comment (lukeed) [reply .eslintrc:21]: eslint borked on `...mapping`. #138 raised Node to 10.x
